diff --git a/test/runtime/samples/each-block-keyed-else/_config.js b/test/runtime/samples/each-block-keyed-else/_config.js
new file mode 100644
--- /dev/null
+++ b/test/runtime/samples/each-block-keyed-else/_config.js
@@ -0,0 +1,37 @@
+export default {
+	props: {
+		animals: ['alpaca', 'baboon', 'capybara'],
+		foo: 'something else'
+	},
+
+	html: `
+		before
+		<p>alpaca</p>
+		<p>baboon</p>
+		<p>capybara</p>
+		after
+	`,
+
+	test({ assert, component, target }) {
+		component.animals = [];
+		assert.htmlEqual(target.innerHTML, `
+			before
+			<p>no animals, but rather something else</p>
+			after
+		`);
+
+		component.foo = 'something other';
+		assert.htmlEqual(target.innerHTML, `
+			before
+			<p>no animals, but rather something other</p>
+			after
+		`);
+
+		component.animals = ['wombat'];
+		assert.htmlEqual(target.innerHTML, `
+			before
+			<p>wombat</p>
+			after
+		`);
+	}
+};
diff --git a/test/runtime/samples/each-block-keyed-else/main.svelte b/test/runtime/samples/each-block-keyed-else/main.svelte
new file mode 100644
--- /dev/null
+++ b/test/runtime/samples/each-block-keyed-else/main.svelte
@@ -0,0 +1,12 @@
+<script>
+	export let animals;
+	export let foo;
+</script>
+
+before
+{#each animals as animal (animal)}
+	<p>{animal}</p>
+{:else}
+	<p>no animals, but rather {foo}</p>
+{/each}
+after
diff --git a/test/runtime/samples/each-block-unkeyed-else-2/_config.js b/test/runtime/samples/each-block-unkeyed-else-2/_config.js
new file mode 100644
--- /dev/null
+++ b/test/runtime/samples/each-block-unkeyed-else-2/_config.js
@@ -0,0 +1,37 @@
+export default {
+	props: {
+		animals: ['alpaca', 'baboon', 'capybara'],
+		foo: 'something else'
+	},
+
+	html: `
+		before
+		<p>alpaca</p>
+		<p>baboon</p>
+		<p>capybara</p>
+		after
+	`,
+
+	test({ assert, component, target }) {
+		component.animals = [];
+		assert.htmlEqual(target.innerHTML, `
+			before
+			<p>no animals, but rather something else</p>
+			after
+		`);
+
+		component.foo = 'something other';
+		assert.htmlEqual(target.innerHTML, `
+			before
+			<p>no animals, but rather something other</p>
+			after
+		`);
+
+		component.animals = ['wombat'];
+		assert.htmlEqual(target.innerHTML, `
+			before
+			<p>wombat</p>
+			after
+		`);
+	}
+};
diff --git a/test/runtime/samples/each-block-unkeyed-else-2/main.svelte b/test/runtime/samples/each-block-unkeyed-else-2/main.svelte
new file mode 100644
--- /dev/null
+++ b/test/runtime/samples/each-block-unkeyed-else-2/main.svelte
@@ -0,0 +1,12 @@
+<script>
+	export let animals;
+	export let foo;
+</script>
+
+before
+{#each animals as animal}
+	<p>{animal}</p>
+{:else}
+	<p>no animals, but rather {foo}</p>
+{/each}
+after
